math/smatrix/inc/Math/SMatrix.icc

```diff
@@ -157,6 +157,12 @@ SMatrix<T,D1,D2,R>& SMatrix<T,D1,D2,R>::operator=(const M&  rhs) {
    return *this;
 }
 
+template <class T, unsigned int D1, unsigned int D2, class R>
+SMatrix<T,D1,D2,R>& SMatrix<T,D1,D2,R>::operator=(const SMatrix<T,D1,D2,R>&  rhs) {
```

Comment:
There is actually an assignment operator defined above. However, it's not picked up for some reason. Does someone know why?

Reply:
@Axel-Naumann just explained me why :) The compiler will always create a default operator unless you implement one with the *exact* signature of the default one. Otherwise it behaves like a free function vs a templated function.

Reply:
To be precise it doesn't have to be *exactly* the same signature, e.g. `Klass(Klass)` would do, or even (evil) `Klass(Klass&)`. There are a couple of allowed versions, basically any function (not function template!) that takes a `Klass()` value and that is not an r-value reference type.